aea/protocols/dialogue/base.py

```diff
@@ -857,66 +851,119 @@ def _basic_validation_non_initial_message(
                 ),
             )
 
-        last_message_id = self.last_message.message_id  # type: ignore
-        if message_id != last_message_id + 1:
-            return (
-                False,
-                "Invalid message_id. Expected {}. Found {}.".format(
-                    last_message_id + 1, message_id
-                ),
-            )
-        if target < 1:
-            return (
-                False,
-                "Invalid target. Expected a value greater than or equal to 1. Found {}.".format(
-                    target
-                ),
+        # it's non initial check, assume there some messages in the dialogue
+        if self.is_empty:
+            return False, "Can not reply cause no messages in the dialogue."
```

Comment:
No need for this check. This method is for non-initial case and is called when self.is_empty() fails.
Also the error message is incorrect.

Reply:
agree, this is non-public method and is_empty check done another place